src/collectors/processcheck/processcheck.py

```diff
@@ -0,0 +1,183 @@
+    :param name: process name
+    :param cmdline: process cmdline
+    :param cfg: the dictionary from processes that describes with the
+        process group we're testing for
+    :return: True if it matches
+    :rtype: bool
+    """
+    for exe_re in cfg['exe']:
+        if exe_re.search(exe):
+            return True
+    for name_re in cfg['name']:
+        if name_re.search(name):
+            return True
+    for cmdline_re in cfg['cmdline']:
+        if cmdline_re.search(' '.join(cmdline)):
+            return True
+    return False
+
+
+def get_value(process, name):
+    result = getattr(process, name)
+    try:
+        return result()
+    except TypeError:
+        return result
+
+
+class ProcessCheckCollector(diamond.collector.Collector):
+
+    def __init__(self, *args, **kwargs):
+        super(ProcessCheckCollector, self).__init__(*args, **kwargs)
+
+        self.hostname = self.get_hostname()
+        self.ttl = self.config['ttl']
+
+        if not netuitive:
+            self.log.error('netuitive import failed. ProcessCheckCollector disabled')
+            self.enabled = False
+            return
+
+        try:
+            self.version = self._get_version()
+            self.api = netuitive.Client(self.config['url'], self.config['api_key'], self.version)
+        except Exception as e:
+            self.log.debug(e)
+
+
+    def process_config(self):
+        super(ProcessCheckCollector, self).process_config()
+        """
+        prepare self.processes, which is a descriptor dictionary in
+        pg_name: {
+            exe: [regex],
+            name: [regex],
+            cmdline: [regex],
+            procs: [psutil.Process]
+        }
+        """
+        self.processes = {}
+        self.processes_info = {}
+        for pg_name, cfg in self.config['process'].items():
+            pg_cfg = {}
+            for key in ('exe', 'name', 'cmdline'):
+                pg_cfg[key] = cfg.get(key, [])
+                if not isinstance(pg_cfg[key], list):
+                    pg_cfg[key] = [pg_cfg[key]]
+                pg_cfg[key] = [re.compile(e) for e in pg_cfg[key]]
+            self.processes[pg_name] = pg_cfg
+            self.processes_info[pg_name] = {}
+
+    def get_default_config_help(self):
+        config_help = super(ProcessCheckCollector,
+                            self).get_default_config_help()
+        config_help.update({
+            'process': ("A subcategory of settings inside of which each "
+                        "collected process has it's configuration"),
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Default settings are:
+            path: 'process'
+        """
+        config = super(ProcessCheckCollector, self).get_default_config()
+        config.update({
+            'path': 'process',
+            'process': {},
+        })
+        return config
+
+    def save_process_info(self, pg_name, process_info):
```

Comment:
What does this method do? I'd expect the process check to just return a boolean. Why do you need to store the up/down value as a counter?